File: 1442-count-triplets-that-can-form-two-arrays-of-equal-xor/1442-count-triplets-that-can-form-two-arrays-of-equal-xor.py
class Solution(object):
    def countTriplets(self, arr):
        n, ans = len(arr) + 1, 0
        prefix = [0] * n

        for i in range(1, n):
            prefix[i] = prefix[i-1] ^ arr[i-1]
        
        for i in range(0, n):
            for j in range(i+1, n):
                if prefix[i] == prefix[j]:
                    ans += (j - i - 1)
        
        return ans
        

File: 1442-count-triplets-that-can-form-two-arrays-of-equal-xor/1442-count-triplets-that-can-form-two-arrays-of-equal-xor.java
class Solution {
    public int countTriplets(int[] arr) {
        int n = arr.length + 1, ans = 0;
        int[] prefix = new int[n];

        for(int i = 1; i<n; i++) {
            prefix[i] = prefix[i-1] ^ arr[i-1];
        } 

        for(int i = 0; i<n; i++) {
            for(int j = i + 1; j<n; j++) {
                if(prefix[i] == prefix[j]) {
                    ans += (j - i - 1);
                }
            }
        }

        return ans;
    }
}